a2z.py

```diff
@@ -0,0 +1,447 @@
+
+       Arguments:
+        - filename (str) : default filename to write to.
+    """
+    # Ensure contributors file and zenodo.json exist
+    if not os.path.exists(zenodo_file):
+        sys.exit("%s does not exist" % zenodo_file)
+
+    if not contributors:
+        contributors = get_contributors(repo)
+
+    # Load the previous contributors, create a lookup
+    metadata = read_json(zenodo_file)
+    lookup = {x["name"]: x for x in metadata.get("creators", [])}
+    for contributor in contributors:
+        user = get_user(contributor["login"])
+        if user["name"] and user["name"] not in lookup:
+            lookup[user["name"]] = {
+                "name": user.get("name") or contributor.get("login"),
+            }
+
+    metadata["creators"] = list(lookup.values())
+    write_json(metadata, zenodo_file)
+    return metadata
+
+
+def update_allcontributors(
+    filename=".all-contributorsrc", thresh=1, ctype="code", contributors=None
+):
+    """Given an existing contributors file, use the GitHub API to retrieve
+       all contributors, and then use subprocess to update the file
+
+       Arguments:
+        - filename (str) : default filename to write to.
+        - thresh (int)   : minimum number of contributions to include
+        - ctype (str)    : the contribution type to add (defaults to code)
+    """
+    if not os.path.exists(filename):
+        sys.exit(
+            "%s does not exist, set --filename or run a2z init to create" % filename
+        )
+
+    # Load the previous contributors, create a lookup
+    metadata = read_json(filename)
+    lookup = {x["login"]: x for x in metadata.get("contributors", [])}
+    repo = "%s/%s" % (metadata["projectOwner"], metadata["projectName"])
+
+    if not contributors:
+        contributors = get_contributors(repo)
+
+    # This doesn't seem to be paginated
+    for contributor in contributors:
+
+        # If they don't meet the threshold, continue
+        if contributor["contributions"] < thresh:
+            continue
+
+        # Skip GitHub bots
+        if contributor["type"] == "Bot" or "[bot]" in contributor["login"]:
+            continue
+
+        # if the username isn't in the lookup, do an extra call to get profile
+        if contributor["login"] not in lookup:
+            user = get_user(contributor["login"])
+            lookup[contributor["login"]] = {
+                "login": contributor["login"],
+                "name": user.get("name", "") or "",
+                "avatar_url": contributor.get("avatar_url"),
+                "profile": user.get("blog", contributor.get("html_url")),
+                "contributions": [ctype],
+            }
+        else:
+            if ctype not in lookup[contributor["login"]]["contributions"]:
+                lookup[contributor["login"]]["contributions"].append(ctype)
+
+    # Update the contributors
+    metadata["contributors"] = list(lookup.values())
+    write_json(metadata, filename)
```

Comment:
I would really prefer to leave ".all-contributorsrc" to all-contributors to deal with.  E.g., although suboptimal since node, not python software - we could call their CLI to cause any changes (e.g. add a contributor, or add a contribution type to an existing one) to this file AND README.md. Otherwise we might interfere with all-contributors bot to operate correctly etc if we introduce some bug in formatting within README, or some unknown field to this json.

Reply:
We don't do anything with the readme, we just add people to "creators" and it's fairly straight forward. The one bug that might come up is if the types of contributions are a limited set we would want to add that to the library.